frontend/src/components/project/ProjectOverview.tsx

```diff
@@ -146,37 +147,22 @@ export default function ProjectOverview({
   const classes = useStyles({});
 
   const texts = getTexts({ page: "project", locale: locale, project: project });
-  /**
-   * Calls endpoint to return a current list
-   * of users that have requested to
-   * join a specific project (i.e. requested membership).
-   *
-   * Note that the response includes a list of requests
-   * (with corresponding request ID), and the users themselves.
-   */
-  async function getMembershipRequests(url_slug) {
-    const resp = await apiRequest({
-      method: "get",
-      url: `/api/projects/${url_slug}/requesters/`,
-    });
 
-    if (!resp?.data?.results) {
-      // TODO: error appropriately here
-    }
-
-    const requestedMembers = resp.data.results;
+  const getHasRequestedToJoin = async () => {
+    try {
+      const resp = await apiRequest({
+        method: "get",
+        url: `/api/projects/${project.url_slug}/have_i_requested_to_join/`,
+        locale: locale,
+        token: token,
+      });
```

Comment:
Opinion: So the more API requests you make to the backend, it adds more time to load the page. I would like you to think, DO I REALLY NEED AN API TO CHECK FOR `have_i_requested_to_join`? Or can it be done with another fetch call? I really think we are at the point where we should question about our how we write our code. I know its a bug that we want to fix but if it can wait for one extra day, I would definitely spend time thinking about the approach here.

Reply:
You're right. I have summarized liking, following and joining in one API call